src/Xamarin.Android.Build.Tasks/Utilities/ITaskItemExtensions.cs

```diff
@@ -25,14 +25,14 @@ public static IEnumerable<XElement> ToXElements (this ICollection<ITaskItem> ite
 		}
 
 		[return: NotNullIfNotNull (nameof (defaultValue))]
-		public static string? GetMetadataOrDefault (this ITaskItem item, string name, string? defaultValue)
+		public static T? GetMetadataOrDefault<T> (this ITaskItem item, string name, T? defaultValue)
 		{
 			var value = item.GetMetadata (name);
 
 			if (string.IsNullOrWhiteSpace (value))
 				return defaultValue;
 
-			return value;
+			return (T?)Convert.ChangeType (value, typeof (T));
```

Comment:
This throws https://learn.microsoft.com/en-us/dotnet/api/system.convert.changetype?view=net-9.0. We should handle that case? and return the default.

Reply:
I think this only throws if the input cannot be converted to the requested type, like say you passed in `tru` and `T` is `bool`.

Passing in an invalid input is a bug that should be fixed, and it feels like we should indeed throw an exception here rather than silently swallow the error and return the default.